app/cloud.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+)
+
+func (a *App) CheckAndSendUserLimitWarningEmails() *model.AppError {
+	if a.Srv().License() != nil && !*a.Srv().License().Features.Cloud {
+		// Not cloud instance, do nothing
+		return nil
+	}
+
+	// subscription, subErr := a.Cloud().GetSubscription()
+	// if subErr != nil {
+	// 	return subErr
+	// }
+
+	// if subscription != nil && subscription.IsPaidTier == "true" {
+	// 	// Paid subscription, do nothing
+	// 	return nil
+	// }
+
+	cloudUserLimit := *a.Config().ExperimentalSettings.CloudUserLimit
+	systemUserCount, _ := a.Srv().Store.User().Count(model.UserCountOptions{})
+	remainingUsers := cloudUserLimit - systemUserCount
+
+	if remainingUsers > 0 {
+		return nil
+	}
+	userOptions := &model.UserGetOptions{
+		Page:     0,
+		PerPage:  100,
+		Role:     model.SYSTEM_ADMIN_ROLE_ID,
+		Inactive: false,
+	}
+	sysAdmins, err := a.GetUsers(userOptions)
+	if err != nil {
+		return err
+	}
+
+	if remainingUsers == -1 {
```

Comment:
I'd put do
```suggestion
	if remainingUsers < 0 {
```
because in case we have some kind of race condition, which I don't know if can happen, and the number of users is -2 then we're not going to send the email

Reply:
This check for -1 is so that the email is only sent one time, and not for every single person added over the limit. I _could_ change this to < 0 but then I'd need to store something somewhere saying "Email has already been sent". I figured that might have been a little overkill for this

Reply:
Ok, please comment the intent so other people knows the reason of it :)

Reply:
Will do!